lib/puppet/type/cisco_bridge_domain.rb

```diff
@@ -0,0 +1,110 @@
+# Manages a Cisco Bridge Domain.
+#
+# March 2016
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_bridge_domain) do
+  @doc = "Manages a Cisco Bridge Domain (BD).
+
+  cisco_bridge_domain {\"<bd>\":
+    ..attributes..
+  }
+
+  <bd> is the id of the bridge domain.
+
+  Example:
+    cisco_bridge_domain {\"1000\":
+      ensure          => present,
+      bd_name         => 'red',
+      fabric_control  => 'false',
+      shutdown        => 'false',
+    }
+  "
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
```

Comment:
Nit: I *think* you can remove this title_patterns method since you have a simple title pattern and the namevar becomes the title. It's not hurting anything by being here though and we have other types that do the same. Take a look at cisco_vtp if you want to remove this.